travis-ci/vmtest/run_selftests.sh

```diff
@@ -43,7 +43,7 @@ fi
 
 cd libbpf/selftests/bpf
 
-test_progs
+[[ "$(uname -m)" == s390x ]] || test_progs
```

Comment:
let's do explicit if and leave a comment why we don't run test_progs on s390x. And yeah, why? I thought the whole point was to run them

Reply:
Currently the trampoline implementation is still missing, but I think having the infrastructure and running test_verifier is nevertheless a good first step. I will add a comment here.

Reply:
see how we do blacklisting/whitelisting for old kernels. We can have black-/whitelists per architecture, if necessary, so you'll be able to enable most tests expect few that you know should not work.

But you are signing up to maintaining this, of course ;) It's the most unpleasant part of libbpf syncs: updating blacklists.

Reply:
Sounds good. I think I'll add a couple `$ARCH` subdirectories under `config/` for that.

Reply:
so if you are going to rework that part, I was always a bit annoyed by a deeply-nested structure of configs. Instead of `configs/blacklist/BLACKLIST-5.5.0` I'd rather have just `BLACKLIST-5.5.0` and `WHITELIST-4.9.0` without directory nesting. So with arch we probably want something like `BLACKLIST-latest.s390x` with fallback (if arch-specific is missing, I don't think we need to merge) to `BLACKLIST-latest` that is common for all arches. WDYT?

Reply:
Ah, I actually did that for configs initially, but then I thought that having directories is better for grouping (e.g. you won't have to look at s390x stuff when working with x86_64), so in my working copy I now have this:

```
├── INDEX
├── s390x
│   ├── blacklist
│   │   └── BLACKLIST-latest
│   └── latest.config
└── x86_64
    ├── blacklist
    │   ├── BLACKLIST-5.5.0
    │   └── BLACKLIST-latest
    ├── latest.config
    └── whitelist
        └── WHITELIST-4.9.0
```

But on the other hand removing blacklist/whitelist nesting makes sense. So what do you think about something like the following?

```
├── INDEX
├── s390x
│   ├── BLACKLIST-latest
│   └── latest.config
└── x86_64
    ├── BLACKLIST-5.5.0
    ├── BLACKLIST-latest
    ├── latest.config
    └── WHITELIST-4.9.0
```

Reply:
I think it's not a lot of configs to need separate per-arch directory, honestly. And I take back what I said about not needing to merge common and per-arch config. I think both are useful because sometimes I have to disable known-bad selftest until it is fixed upstream (and merged back into bpf-next). So having a common place where I need to add one entry once is good. But we don't have to really merge them, it's easier to add support for specifying multiple whitelists and blacklists to test_progs.

So overall, I still think .arch suffix and flat configs directory is better.